service/interpreter/workflowImpl.go

```diff
@@ -487,6 +486,23 @@ func checkClosingWorkflow(
 	return
 }
 
+func DrainReceivedButUnprocessedInternalChannelsFromStateApis(
+	ctx UnifiedContext, provider WorkflowProvider, versioner *GlobalVersioner,
+) error {
+	if versioner.IsAfterVersionOfYieldOnConditionalComplete() {
+		// Just yield, by waiting on an empty lambda, nothing else.
+		// It will let other workflow threads/coroutines to run.
+		// This will drain the messages published from state APIs.
+		// NOTE that this is extremely tricky in Cadence/Temporal programming model.
+		// Read more: https://stackoverflow.com/questions/71356668/how-does-multi-threading-works-in-cadence-temporal-workflow
+		//https://docs.temporal.io/encyclopedia/go-sdk-multithreading
+		return provider.Await(ctx, func() bool {
+			return true
+		})
+	}
```

Comment:
I'm gonna trust this works as expected. Having tests would be beneficial and make it easier to understand and verify. Hope we can cover multi-threading soon so I can get a better grasp on it.